Update Graph Module specification (#1031)
blockprotocol/blockprotocol: apps/site/src/_pages/docs/1_developing-blocks.mdx

- `depths`: which `edges` were followed from the `roots` when resolving the query, and how far
- `edges`: connections between things in the graph. For example, a 'link entity' connects it and two other entities via `hasRightEntity`
  and `hasLeftEntity` edges (conceptually, the link entity is in the middle with an entity on its left and on its right)

> - `vertices`: each thing (`vertex`, or `node`) which was encountered when starting from the roots and following the specified `edges

Author: I've opted to remove `node` here because I think introducing yet another term is confusing (even if "node" is fairly common in describing graphs), replaced it with "element"

to the specified`depths`.

> A `Subgraph` may also be rooted at _types_, and it may also contain types in its _vertices_, as well as have _edges_

Author: I've completely removed this section as we don't mention that the `Subgraph` can currently have types within it anywhere else in the spec. We can re-add this section once we extend the implementation with that.=

There are messages for exploring the data available in the embedding application:

> - `aggregateEntities` allows you to build a query to filter the entities in the data store.

Author: ```suggestion
- `aggregateEntities` allows you submit a query which filters the entities in the data store.
```
